src/mame/korg/korgtriton.cpp

```diff
@@ -0,0 +1,210 @@
+// license:BSD-3-Clause
+// copyright-holders:Antonio "Willy" Malara
+
+/*
+Korg TRITON skeleton driver
+
+CPU: SH2 SH7043 / SH7045
+FLASH: 2 * MX29F1610
+LCD controller: M66271FP
+Floppy disk controller: HD63266
+Serial Interface: uPD71051
+SCSI controller: MB86604L
+Sound: 2 * "TGL96" MB87F1710-PFV-S (proprietary)
+
+This driver runs the embedded firmware up until it needs to
+relocate and run the main application. The screen remains
+blank for all this time.
+*/
+
+
+#include "emu.h"
+#include "screen.h"
+
+#include "cpu/sh/sh7042.h"
+#include "machine/i8251.h"
+
+namespace {
+
+    class korgtriton_state : public driver_device
+    {
+    public:
+        korgtriton_state(const machine_config &mconfig, device_type type, const char *tag)
+        : driver_device(mconfig, type, tag)
+        , m_maincpu(*this, "maincpu")
+        , m_ram(*this, "ram")
+        , m_lcdcm(*this, "lcdcm")
+        , m_lcdcio(*this, "lcdcio")
+        , m_screen(*this, "screen")
+        , m_scu(*this, "scu")
+        { }
+
+        void korgtriton(machine_config &config);
+
+    private:
+        required_device<sh7043_device> m_maincpu;
+        required_shared_ptr<u32> m_ram;
+        required_shared_ptr<u32> m_lcdcm;
+        required_shared_ptr<u32> m_lcdcio;
+        required_device<screen_device> m_screen;
+        required_device<i8251_device> m_scu;
+
+        void map(address_map &map) ATTR_COLD;
+
+        virtual void machine_start() override ATTR_COLD;
+        virtual void machine_reset() override ATTR_COLD;
+
+        uint32_t screen_update(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect);
+
+        u32 pa_r();
+        void pa_w(u32 data);
+
+        u32 pe_r();
+        void pe_w(u32 data);
+
+        void scu_txrdy_w(int state);
+        void scu_rxrdy_w(int state);
+
+        u8 scu_read(offs_t offs) {
+            int res = 0;
+
+            if (offs == 0) {
+                static int i = 0;
+                static u8 pd[] = { 0x66, 0x31 };
+                res = pd[i++ % 2];
+            }
+
+            if (offs == 1) {
+                static int hack = 0;
+                res = hack++;
+            }
+
+            printf("scu_read: %08x -> %02x\n", offs, res);
+            return res;
+        }
+
+        void scu_write(offs_t offs, u8 data) {
+            printf("scu_write: %08x %02x\n", offs, data);
+        }
```

Comment:
Please don’t put non-trivial function bodies inline at declaration.  It makes class declarations harder to read.

Please don’t add non-`const` function statics.  They break save states, make debugging hard, and are generally a bug waiting to happen.

Please check `machine().side_effects_disabled()` before doing things with side effects in a read handler.